agent/taskresource/volume/dockervolume.go

```diff
@@ -32,6 +34,7 @@ type VolumeResource struct {
 	desiredStatusUnsafe VolumeStatus
 	knownStatusUnsafe   VolumeStatus
 	client              dockerapi.DockerClient
+	ctx                 context.Context
```

Comment:
Where is this being initialized?

Reply:
It has not been initialized yet. That will be done when we create a new `VolumeResource`

Reply:
Not all resources might need `Context` for `Create`/`Cleanup`. Hence added as part of the resource struct.

Reply:
> It has not been initialized yet. That will be done when we create a new `VolumeResource` 

What's stopping you from initializing it in this PR?